libs/@local/hash-graph-sdk/typescript/src/entity.ts

```diff
@@ -463,6 +470,32 @@ export const flattenPropertyMetadata = (
   return flattened;
 };
 
+export const getClosedMultiEntityTypesFromResponse = (
+  response: GetEntitySubgraphResponse | GetEntitiesResponse,
+  entityTypesIds: [VersionedUrl, ...VersionedUrl[]],
+): ClosedMultiEntityType | undefined => {
+  if (!response.closedMultiEntityTypes) {
+    return;
+  }
+
+  const [firstEntityTypeId, ...restEntityTypesIds] = entityTypesIds.sort();
+  let currentClosedMultiEntityTypeMap =
+    response.closedMultiEntityTypes[firstEntityTypeId];
+
+  for (const id of restEntityTypesIds) {
+    if (!currentClosedMultiEntityTypeMap?.inner) {
+      return;
+    }
+    currentClosedMultiEntityTypeMap = currentClosedMultiEntityTypeMap.inner[id];
+  }
```

Comment:
Would like to understand what is going on here, I would have thought a closed multi-entity type would be keyed under some combination of all its typeIds but here we have items keyed by a single typeId and are returning whatever the item in the map with the last entityTypeId in the list is (unless one is missing, in which case we return nothing). Why is it like this, and what happens if there are multiple multi-type entities which have the same entityTypeId amongst their types?

Reply:
Oh, it's a nested map which is nested to however many levels there are numbers of types for the entity... is this what we want long-term, or is it just temporary? I thought we'd just sort and concatenate the typeIds or something. Is there a benefit to doing it this way?

Reply:
I don't know if we would keep it longer term. This is the reason why I have "hidden" the implementation behind the `response` object. I'm currently writing a few comments to explain this more. Also, I try to rewrite it to use `reduce` instead.
The reason why I used this is because it's easier to build and does not need to reallocate strings when concatenating strings. It can be iteratively be built in the graph as it's layered.